libraries/shared/storage/fetcher/geth_rpc_storage_fetcher.go

```diff
@@ -66,45 +61,177 @@ func (fetcher GethRpcStorageFetcher) FetchStorageDiffs(out chan<- types.RawDiff,
-						diff.HashedAddress.Hex(), diff.BlockHeight, diff.StorageKey.Hex(), diff.StorageValue.Hex())
-					if formatErr != nil {
-						errs <- formatErr
-					}
-
-					out <- diff
-				}
+func (fetcher GethRpcStorageFetcher) handleDiffsFromOldGethPatch(payload filters.Payload, out chan<- types.RawDiff, errs chan<- error) {
+	stateDiff, decodeErr := fetcher.decodeStateDiffRlpFromPayloadForOldPatches(payload)
+	if decodeErr != nil {
+		errs <- decodeErr
+	}
+
+	for _, account := range stateDiff.UpdatedAccounts {
+		logrus.Trace(fmt.Sprintf("iterating through %d Storage values on account", len(account.Storage)))
+
+		for _, accountStorage := range account.Storage {
+			rawDiff, formatErr := types.FromOldGethStateDiff(account, stateDiff, accountStorage)
+			if formatErr != nil {
+				errs <- formatErr
 			}
+
+			logrus.Tracef("adding storage diff to out channel. keccak of address: %v, block height: %v, storage key: %v, storage value: %v",
+				rawDiff.HashedAddress.Hex(), rawDiff.BlockHeight, rawDiff.StorageKey.Hex(), rawDiff.StorageValue.Hex())
+			out <- rawDiff
 		}
+	}
+}
 
+func (fetcher GethRpcStorageFetcher) handleDiffsFromNewGethPatchWithService(payload filters.Payload, out chan<- types.RawDiff, errs chan<- error) {
+	stateDiff, decodeErr := fetcher.decodeStateDiffRlpFromPayloadForOldPatches(payload)
+	if decodeErr != nil {
+		errs <- decodeErr
+	}
+
+	for _, account := range stateDiff.UpdatedAccounts {
+		logrus.Trace(fmt.Sprintf("iterating through %d Storage values on account", len(account.Storage)))
+
+		for _, accountStorage := range account.Storage {
+			rawDiff, formatErr := types.FromNewGethStateDiff(account, stateDiff, accountStorage)
+			if formatErr != nil {
+				errs <- formatErr
+			}
+
+			logrus.Tracef("adding storage diff to out channel. keccak of address: %v, block height: %v, storage key: %v, storage value: %v",
+				rawDiff.HashedAddress.Hex(), rawDiff.BlockHeight, rawDiff.StorageKey.Hex(), rawDiff.StorageValue.Hex())
+			out <- rawDiff
+		}
 	}
 }
 
-func (fetcher GethRpcStorageFetcher) formatDiff(account statediff.AccountDiff, stateDiff *statediff.StateDiff, storage statediff.StorageDiff) (types.RawDiff, error) {
-	if fetcher.gethVersion == OldGethPatch {
-		return types.FromOldGethStateDiff(account, stateDiff, storage)
+func (fetcher GethRpcStorageFetcher) handleDiffsFromNewGethPatchWithFilter(payload filters.Payload, out chan<- types.RawDiff, errs chan<- error) {
+	var stateDiff filters.StateDiff
+	decodeErr := rlp.DecodeBytes(payload.StateDiffRlp, &stateDiff)
+	if decodeErr != nil {
+		errs <- decodeErr
+	}
+
+	for _, account := range stateDiff.UpdatedAccounts {
+		logrus.Trace(fmt.Sprintf("iterating through %d Storage values on account", len(account.Storage)))
+
+		for _, accountStorage := range account.Storage {
+			rawDiff, formatErr := types.FromNewGethStateDiff(account, &stateDiff, accountStorage)
```

Comment:
Why does `stateDiff` need to be referenced by pointer here but not on line 111?

Reply:
On line 111, the `stateDiff` that's returned from `fetcher.decodeStateDiffRlpFromPayloadForOldPatches` is also a pointer. I didn't wrap decoding the new geth patch diffs in a new method like I did for the old geth patch diffs, since it's just a one liner and doesn't require decoding into an intermediary `StateDiffOldPatch` structure.